# HG changeset patch
# User Rik <rik@octave.org>
# Date 1515456807 28800
#      Mon Jan 08 16:13:27 2018 -0800
# Node ID 5a10409695b7505042b5d8b8cd76e8840858cdaf
# Parent  06e22134d81ac70dc120155f396822d78adb7ccc
Open GUI external editor at beginning of file (bug #52811).

* external-editor-interface.cc (call_custom_editor ): If line number is not
set (-1) then use line of 0 (beginning of file).

diff --git a/libgui/src/external-editor-interface.cc b/libgui/src/external-editor-interface.cc
--- a/libgui/src/external-editor-interface.cc
+++ b/libgui/src/external-editor-interface.cc
@@ -36,16 +36,18 @@ external_editor_interface::external_edit
 { }
 
 // Calling the external editor
 bool
 external_editor_interface::call_custom_editor (const QString& file, int line)
 {
   if (line > -1)  // check for a specific line (debugging)
     return true;  // yes: do not open a file in external editor
+  else
+    line = 0;     // no: start external editor at beginning of file
 
   QString editor = external_editor ();
   if (editor.isEmpty ())
     return true;
 
   // replace macros
   editor.replace ("%f", file);
   editor.replace ("%l", QString::number (line));
